barretenberg/cpp/src/barretenberg/vm2/constraining/relations/success_copy.test.cpp

```diff
@@ -0,0 +1,75 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+
+#include "barretenberg/vm2/constraining/flavor_settings.hpp"
+#include "barretenberg/vm2/constraining/testing/check_relation.hpp"
+#include "barretenberg/vm2/generated/relations/execution.hpp"
+#include "barretenberg/vm2/testing/macros.hpp"
+#include "barretenberg/vm2/tracegen/test_trace_container.hpp"
+
+namespace bb::avm2::constraining {
+namespace {
+
+using tracegen::TestTraceContainer;
+using FF = AvmFlavorSettings::FF;
+using C = Column;
+using execution = bb::avm2::execution<FF>;
+
+TEST(SuccessCopyConstrainingTest, SuccessCopyTrue)
+{
+    TestTraceContainer trace({
+        { { C::precomputed_first_row, 1 } },
+        { { C::execution_sel, 1 },
+          { C::execution_sel_success_copy, 1 },
+          { C::execution_register_0_, /*true=*/1 },
+          { C::execution_mem_tag_reg_0_, /*U1=*/1 },
```

Comment:
I've changed many of these to `static_cast<uint8_t>(MemoryTag::U1)` which I think is better. Same in other tests. No need to hardcode numbers.